src/lib/transport/ip/netif_tx.c

```diff
@@ -365,28 +365,6 @@ void __ci_netif_send(ci_netif* netif, ci_ip_pkt_fmt* pkt)
    */
   LOG_NT(log("%s: ENQ id=%d", __FUNCTION__, OO_PKT_FMT(pkt)));
   __ci_netif_dmaq_put(netif, dmaq, pkt);
-
- done:
-
-  /* Poll every now and then to ensure we keep up with completions.  If we
-   * don't do this then we can ignore completions for so long that we start
-   * putting stuff on the overflow queue when we don't really need to.
-   */
-  if( netif->state->send_may_poll ) {
```

Comment:
`send_may_poll` is used from UDP code, and it seems sensible.  UDP send does not happen from the depths of other calls.

I considered checking `in_poll` field instead of removing this code entirely.  As we see in the second issue, poll is not the only recursion we'd like to avoid; sending should not recurse as well.  So it looks safer to remove this completely.

Additionally, the idea of this code is performance, and I do not believe it really serves this purpose in any realistic scenario.  My tests support this.  We have ACKs in TCP, so we are forced to poll the stack regularly.

NB If I were you I wouldn't merge this until you fix a recent regression with `ci_assert((nic_i)< (oo_stack_intf_max(ni))) on loopback ` from 2f4bbe561d8.  Just to make future bisecting easier.

Reply:
The assertion with `send_may_poll` in the UDP code was borderline "used" in my eyes, but sure let's keep it if you think it's worth it.

> I wouldn't merge this until you fix a recent regression

Thanks and @jfeather-amd has a pending fix for that. 👍

Reply:
> The assertion with `send_may_poll` in the UDP code was borderline "used" in my eyes, but sure let's keep it if you think it's worth it.

Let me re-check.  I remember it was a real use...

Reply:
`ci_udp_sendmsg_send()`:
```c
  ci_assert_equal(ni->state->send_may_poll, 0);
  ni->state->send_may_poll = ci_netif_may_poll(ni);

....

  ni->state->send_may_poll = 0;
```
And I believe it is the correct and useful usage.

Reply:
Totally, and as I said before, if you think the assertion is a good enough case, let's keep it.